src/test/java/org/broadinstitute/hellbender/cmdline/PicardProgramExecutorIntegrationTest.java

```diff
@@ -0,0 +1,64 @@
+package org.broadinstitute.hellbender.cmdline;
+
+import org.broadinstitute.hellbender.CommandLineProgramTest;
+import org.broadinstitute.hellbender.utils.test.IntegrationTestSpec;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+
+// Test that we can successfully run Picard tools.
+public final class PicardProgramExecutorIntegrationTest extends CommandLineProgramTest {
+
+    // Use the Picard tool "NormalizeFasta" as a simple test case.
+    private static final File TEST_DATA_PATH = new File(getTestDataDir(), "picard/fastq/NormalizeFasta");
+
+    @Override
+    public String getTestedClassName() {
+        return picard.reference.NormalizeFasta.class.getSimpleName();
+    }
+
+    @Override
+    public List<String> injectDefaultVerbosity(final List<String> args) {
+        // override/suppress GATK-specific argument injection
+        return args;
+    }
+
+    @Test
+    public void testPicardNormalizeFasta() throws IOException {
+        final File input = new File(TEST_DATA_PATH, "testfasta.fasta");
+        final File expectedFile = new File(TEST_DATA_PATH, "testFASTA_WS_4.fasta");
+        final File outfile = createTempFile("normalized", ".fasta");
+
+        final String[] args = {
+                "-I", input.getAbsolutePath(),
+                "-O", outfile.getAbsolutePath(),
+                "--TRUNCATE_SEQUENCE_NAMES_AT_WHITESPACE", "TRUE",
+                "--LINE_LENGTH", "5",
+        };
+
+        Assert.assertEquals(runCommandLine(args), 0);
+
+        IntegrationTestSpec.assertEqualTextFiles(outfile, expectedFile);
+    }
+
+    @Test
+    public void testPicardNormalizeFastaWithBadArgs() throws IOException {
+        final File input = new File(TEST_DATA_PATH, "testfasta.fasta");
+        final File outfile = createTempFile("normalized", ".fasta");
+
+        // use GATK-style lower case names, which are rejected by Picard
+        final String[] args = {
+                "--input", input.getAbsolutePath(),
+                "--output", outfile.getAbsolutePath(),
+                "--TRUNCATE_SEQUENCE_NAMES_AT_WHITESPACE", "TRUE",
+                "--LINE_LENGTH", "5",
+        };
+
+        Assert.assertNotEquals(runCommandLine(args), 0);
```

Comment:
Why do we get a bad return value here instead of an exception bubbling up?  If the args are no good shouldn't barclay be throwing an exception?

Reply:
Picard swallows Barclay exceptions and propagates them as a return code since it also has to work with the legacy parser, which uses return codes instead of exceptions.